lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -474,17 +493,20 @@ bool TypeLoweringVisitor::lowerProducer(
       loweredName += field.suffix;
     }
     bool hasDontTouch = false;
+    StringAttr sName;
+    if (innerSym)
+      sName = StringAttr::get(context, innerSym.getValue() + loweredName);
+    else
+      sName = StringAttr::get(context, loweredName);
```

Comment:
do you ever have to synthesize a new symbol name if there wasn't already a symbol name?

Reply:
LowerTypes might synthesize a new name, if the `DontTouch` is present on a bundle field only.
In that case, the symbol might not be present on the original operation.